### PR TITLE
Add test - Quick view product on FO

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.7.x
| Description?      | Add some tests on quick view modal FO( Add to cart, Change quantity, Share links, Display of the product, Close modal, Change combination on the popup, Select color on hover on the product list, change image) 
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | o
| Fixed ticket?     | no
| How to test?      | TEST_PATH="functional/FO/08_productPage/01_quickView.js" URL_FO=yourShopURL npm run specific-test
| Possible impacts? | no impacts

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23238)

### CODE REVIEW

File: themes/classic/modules/ps_shoppingcart/modal.tpl [hunk 1]
```diff
@@ -56,9 +56,9 @@
               {else}
                 <p class="cart-products-count">{l s='There is %product_count% item in your cart.' sprintf=['%product_count%' =>$cart.products_count] d='Shop.Theme.Checkout'}</p>
               {/if}
-              <p><span class="label">{l s='Subtotal:' d='Shop.Theme.Checkout'}</span>&nbsp;<span class="value">{$cart.subtotals.products.value}</span></p>
+              <p><span class="label">{l s='Subtotal:' d='Shop.Theme.Checkout'}</span>&nbsp;<span class="subtotals" class="value">{$cart.subtotals.products.value}</span></p>
```
Comment (boubkerbribri) [inline themes/classic/modules/ps_shoppingcart/modal.tpl]:
```suggestion
              <p><span class="label">{l s='Subtotal:' d='Shop.Theme.Checkout'}</span>&nbsp;<span class="subtotals value">{$cart.subtotals.products.value}</span></p>
```

File: themes/classic/modules/ps_shoppingcart/modal.tpl [hunk 2]
```diff
@@ -56,9 +56,9 @@
...
               {if $cart.subtotals.shipping.value}
-                <p><span>{l s='Shipping:' d='Shop.Theme.Checkout'}</span>&nbsp;<span class="value">{$cart.subtotals.shipping.value} {hook h='displayCheckoutSubtotalDetails' subtotal=$cart.subtotals.shipping}</span></p>
+                <p><span>{l s='Shipping:' d='Shop.Theme.Checkout'}</span>&nbsp;<span class='shipping'class="value">{$cart.subtotals.shipping.value} {hook h='displayCheckoutSubtotalDetails' subtotal=$cart.subtotals.shipping}</span></p>
```
Comment (boubkerbribri) [inline themes/classic/modules/ps_shoppingcart/modal.tpl]:
```suggestion
                <p><span>{l s='Shipping:' d='Shop.Theme.Checkout'}</span>&nbsp;<span class="shipping value">{$cart.subtotals.shipping.value} {hook h='displayCheckoutSubtotalDetails' subtotal=$cart.subtotals.shipping}</span></p>
```

File: themes/classic/templates/checkout/_partials/cart-detailed-product-line.tpl [hunk 3]
```diff
@@ -66,7 +66,7 @@
     <br/>
 
     {foreach from=$product.attributes key="attribute" item="value"}
-      <div class="product-line-info">
+      <div id={$attribute} class="product-line-info">
```
Comment (boubkerbribri) [inline themes/classic/templates/checkout/_partials/cart-detailed-product-line.tpl]:
```suggestion
      <div class="product-line-info attribute-{$attribute|lower}-info">
```

Comment (boubkerbribri) [reply themes/classic/templates/checkout/_partials/cart-detailed-product-line.tpl]: You should use class instead, to avoid duplicated ids

Comment (nesrineabdmouleh) [reply themes/classic/templates/checkout/_partials/cart-detailed-product-line.tpl]: Done thx :+1:

**Comment by Progi1984** [comment]
@nesrineabdmouleh Sanity seems to be broken.

File: tests/UI/pages/BO/catalog/products/add.js [hunk 4]
```diff
@@ -95,6 +97,12 @@ class AddProduct extends BOBasePage {
    */
   async setBasicSetting(page, productData) {
     await this.setValue(page, this.productNameInput, productData.name);
+    if (productData.coverImage != null) {
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/products/add.js]:
```suggestion
    if (productData.coverImage !== null) {
```

File: tests/UI/pages/BO/catalog/products/add.js [hunk 5]
```diff
@@ -95,6 +97,12 @@ class AddProduct extends BOBasePage {
...
+      await this.uploadFilePath(page, this.productImageDropZoneDiv, productData.coverImage);
+    }
+    if (productData.thumbImage != null) {
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/products/add.js]:
```suggestion
    if (productData.thumbImage !== null) {
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

**Review by Progi1984 — APPROVED** [review]

**Review by matks — DISMISSED** [review]

**Review by Progi1984 — CHANGES_REQUESTED** [review]
## WAIT FOR 1772 RELEASE

**Review by Progi1984 — APPROVED** [review]

**Review by Progi1984 — CHANGES_REQUESTED** [review]
## WAIT FOR 1772 RELEASE

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**